Tested running on https://github.com/benbusby/colorstorm/blob/main/examples/earthbound-themes.json as `colorstorm earthbound-themes.json`. Unfortunately, it looks like the changes from https://github.com/benbusby/colorstorm/pull/15 weren't thoroughly tested.

## Before this PR:

### `aarch64-darwin`
<details>
  <summary>Zig 0.9 build failure</summary>

       > -- Installing: /nix/store/d1xamfvzwgpikfgd9xam14vhfm54543p-zig-0.9.1/lib
       > thread 802309 panic: Darwin is handled separately via std.zig.system.darwin module
       > Unable to dump stack trace: debug info stripped
       > ::
       > :: ERROR: Subprocess aborted
       > :: (execute_process)
       > ::
       > :: argv: /tmp/nix-build-zig-0.9.1.drv-0/source/build/zig build --zig-lib-dir /tmp/nix-build-zig-0.9.1.drv-0/source/lib -Dlib-files-only --prefix /nix/store/d1xamfvzwgpikfgd9xam14vhfm54543p-zig-0.9.1 -Dconfig_h=/tmp/nix-build-zig-0.9.1.drv-0/source/build/config.h install
       > ::
       > :: argv[0]: /tmp/nix-build-zig-0.9.1.drv-0/source/build/zig
       > :: argv[1]: build
       > :: argv[2]: --zig-lib-dir
       > :: argv[3]: /tmp/nix-build-zig-0.9.1.drv-0/source/lib
       > :: argv[4]: -Dlib-files-only
       > :: argv[5]: --prefix
       > :: argv[6]: /nix/store/d1xamfvzwgpikfgd9xam14vhfm54543p-zig-0.9.1
       > :: argv[7]: -Dconfig_h=/tmp/nix-build-zig-0.9.1.drv-0/source/build/config.h
       > :: argv[8]: install
       > ::
       > CMake Error at /tmp/nix-build-zig-0.9.1.drv-0/source/cmake/install.cmake:36 (message):
       > Call Stack (most recent call first):
       >   cmake_install.cmake:67 (include)
       >
       >
       > make: *** [Makefile:100: install] Fehler 1

</details>

### `x86_64-linux`

<details>
  <summary>Build and execution successful</summary>

```
-- vim
   ./colorstorm-out/vim/colors/earthbound.vim
   ./colorstorm-out/vim/colors/moonside.vim
   ./colorstorm-out/vim/colors/threed.vim
   ./colorstorm-out/vim/colors/fire-spring.vim
   ./colorstorm-out/vim/colors/dusty-dunes.vim
   ./colorstorm-out/vim/colors/magicant.vim
   ./colorstorm-out/vim/colors/cave-of-the-past.vim
   ./colorstorm-out/vim/colors/devils-machine.vim
-- vscode
   ./colorstorm-out/vscode/themes/earthbound.json
   ./colorstorm-out/vscode/themes/moonside.json
   ./colorstorm-out/vscode/themes/threed.json
   ./colorstorm-out/vscode/themes/fire-spring.json
   ./colorstorm-out/vscode/themes/dusty-dunes.json
   ./colorstorm-out/vscode/themes/magicant.json
   ./colorstorm-out/vscode/themes/cave-of-the-past.json
   ./colorstorm-out/vscode/themes/devils-machine.json
-- sublime
   ./colorstorm-out/sublime/earthbound.tmTheme
   ./colorstorm-out/sublime/moonside.tmTheme
   ./colorstorm-out/sublime/threed.tmTheme
   ./colorstorm-out/sublime/fire-spring.tmTheme
   ./colorstorm-out/sublime/dusty-dunes.tmTheme
   ./colorstorm-out/sublime/magicant.tmTheme
   ./colorstorm-out/sublime/cave-of-the-past.tmTheme
   ./colorstorm-out/sublime/devils-machine.tmTheme
-- atom
   ./colorstorm-out/atom/themes/earthbound-syntax/colors.less
   ./colorstorm-out/atom/themes/moonside-syntax/colors.less
   ./colorstorm-out/atom/themes/threed-syntax/colors.less
   ./colorstorm-out/atom/themes/fire-spring-syntax/colors.less
   ./colorstorm-out/atom/themes/dusty-dunes-syntax/colors.less
   ./colorstorm-out/atom/themes/magicant-syntax/colors.less
   ./colorstorm-out/atom/themes/cave-of-the-past-syntax/colors.less
   ./colorstorm-out/atom/themes/devils-machine-syntax/colors.less
-- iterm2
   ./colorstorm-out/iterm2/earthbound.itermcolors
   ./colorstorm-out/iterm2/moonside.itermcolors
   ./colorstorm-out/iterm2/threed.itermcolors
   ./colorstorm-out/iterm2/fire-spring.itermcolors
   ./colorstorm-out/iterm2/dusty-dunes.itermcolors
   ./colorstorm-out/iterm2/magicant.itermcolors
   ./colorstorm-out/iterm2/cave-of-the-past.itermcolors
   ./colorstorm-out/iterm2/devils-machine.itermcolors
```

</details>

## After this PR:

### `aarch64-darwin`
<details>
  <summary>Execution fails with segmentation fault</summary>

```
-- vim
Segmentation fault at address 0x100af8010
Panicked during a panic. Aborting.
Abort trap: 6
```

</details>

### `x86_64-linux`
<details>
  <summary>Execution fails with segmentation fault</summary>

```
-- vim
Segmentation fault at address 0x7f71652ac010
???:?:?: 0x101d94a in ??? (exe)
Unwind information for `exe:0x101d94a` was not available, trace may be incomplete

[1]    1674128 abort (core dumped)
```

</details>

--- a/pkgs/by-name/co/colorstorm/0001-fix-segfault.patch
+++ b/pkgs/by-name/co/colorstorm/0001-fix-segfault.patch
@@ -1,0 +1,12 @@
+diff --git a/src/templator.zig b/src/templator.zig
+index 5630a04..0dc8ca7 100644
+--- a/src/templator.zig
++++ b/src/templator.zig
+@@ -77,7 +77,6 @@ pub fn parse_themes(f: std.fs.File) ![]Theme {
+     }
+ 
+     const parsed = try std.json.parseFromSlice([]Theme, a, list.items, .{});
+-    defer parsed.deinit();
+     const themes = parsed.value;
+ 
+     return themes;

--- a/pkgs/by-name/co/colorstorm/package.nix
+++ b/pkgs/by-name/co/colorstorm/package.nix
@@ -2,22 +2,31 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  zig_0_9,
+  zig_0_13,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "colorstorm";
-  version = "2.0.0";
+  # last tagged release is three years old and requires outdated Zig 0.9
+  # new release requested in: https://github.com/benbusby/colorstorm/issues/16
+  version = "2.0.0-unstable-2025-01-17";
 
   src = fetchFromGitHub {
     owner = "benbusby";
     repo = "colorstorm";
-    rev = "v${finalAttrs.version}";
-    hash = "sha256-6+P+QQpP1jxsydqhVrZkjl1gaqNcx4kS2994hOBhtu8=";
+    rev = "e645c4293fb5f72968038dac99e0b8dab3db194f";
+    hash = "sha256-6D+aNcjJksv7E9RJB9fnzgzvGoUPXV4Shz5wLu5YHtg=";
   };
 
+  patches = [
+    # Fixes a use-after-free segfault.
+    # See https://github.com/benbusby/colorstorm/pull/15#discussion_r1930406581
+    # and upstream PR https://github.com/NixOS/nixpkgs/pull/377279
+    ./0001-fix-segfault.patch
+  ];
+
   nativeBuildInputs = [
-    zig_0_9.hook
+    zig_0_13.hook
   ];
 
   meta = {
@@ -25,7 +34,7 @@ stdenv.mkDerivation (finalAttrs: {
     homepage = "https://github.com/benbusby/colorstorm";
     license = lib.licenses.mit;
     maintainers = [ ];
-    inherit (zig_0_9.meta) platforms;
+    inherit (zig_0_13.meta) platforms;
     mainProgram = "colorstorm";
   };
 })

